drakrun/drakrun/lib/postprocessing/generate_report.py

```diff
@@ -0,0 +1,187 @@
+                pkey = process_key(occurrence["ppid"], occurrence["pid"])
+                processes[pkey]["ttps"].append(ttp)
+
+
+def parse_memdumps(processes: Dict, memdumps_file: Path) -> None:
+    # This method parses the memdump.log file and appends all memory dump
+    # information into the appropriate process in the report
+    with memdumps_file.open("r") as f:
+        for line in f:
+            memdump: Dict = orjson.loads(line)
+            pkey = process_key(memdump["PPID"], memdump["PID"])
+            processes[pkey]["memdumps"].append(
+                {
+                    "reason": memdump["DumpReason"],
+                    "addr": memdump["DumpAddr"],
+                    "size": memdump["DumpSize"],
+                    "filename": memdump["DumpFilename"],
+                    "count": memdump["DumpsCount"],
+                }
+            )
+
+
+def parse_processtree(processtree_file: Path) -> List[Dict]:
+    # This method extracts all the processes and their associated information
+    # from the process_tree.json file.
+    def rec(processes: List[Dict], parent=0) -> Iterator[Dict]:
+        # This is a helper recursive function that parses the process tree
+        for process in processes:
+            yield {
+                "pid": process["pid"],
+                "ppid": parent,
+                "procname": process["procname"],
+                "args": process["args"],
+                "ts_from": epoch_to_timestring(process["ts_from"]),
+                "ts_to": epoch_to_timestring(process["ts_to"]),
+                "children": [
+                    process_key(process["pid"], child["pid"])
+                    for child in process["children"]
+                ],
+                "api_calls": [],  # to be filled later by parse_apimon()
+                "ttps": [],  # to be filled later by parse_ttps()
+                "memdumps": [],  # to be filled later by parse_memdumps()
+            }
+            yield from rec(process["children"], parent=process["pid"])
+
+    with processtree_file.open("r") as f:
+        processtree = orjson.loads(f.read())
+
+    return {
+        process_key(process["ppid"], process["pid"]): process
+        for process in rec(processtree)
+    }
+
+
+def get_metadata(analysis_dir: Path) -> Dict:
+    # Currently, all metadata is contained in the metadata.json file
+    return parse_metadata(analysis_dir / "metadata.json")
+
+
+def get_processes(analysis_dir: Path) -> Dict:
+    # generate a dictionary of indexed processes
+    processes = parse_processtree(analysis_dir / "process_tree.json")
+    # parse api calls into the indexed process dictionary
+    parse_apimon(processes, analysis_dir / "apimon.log")
+    # parse ttps into the indexed process dictionary
+    parse_ttps(processes, analysis_dir / "ttps.json")
+    # parse memory dumps log into the indexed process dictionary
+    parse_memdumps(processes, analysis_dir / "memdump.log")
+
+    return processes
```

Comment:
Code should handle the situation where each of these files don't exist (for example, TTP information is missing) by a progressive degradation (so for example process tree will be available even if TTP data is missing)